pkg/gardenlet/controller/federatedseed/seed_control.go

```diff
@@ -105,25 +105,16 @@ func (c *Controller) addSeedToQueue(obj interface{}) {
 	c.seedQueue.Add(key)
 }
 
-func (c *Controller) seedUpdate(oldObj, newObj interface{}) {
-	var (
-		newSeedObj = newObj.(*gardencorev1beta1.Seed)
-		oldSeedObj = oldObj.(*gardencorev1beta1.Seed)
-	)
+func (c *Controller) seedUpdate(_, newObj interface{}) {
+	var newSeedObj = newObj.(*gardencorev1beta1.Seed)
 
 	newSeedBootstrappedCondition := helper.GetCondition(newSeedObj.Status.Conditions, gardencorev1beta1.SeedBootstrapped)
-	oldSeedBootstrappedCondition := helper.GetCondition(oldSeedObj.Status.Conditions, gardencorev1beta1.SeedBootstrapped)
-
-	if seedChangedToBootstrapped(newSeedBootstrappedCondition, oldSeedBootstrappedCondition) {
+	_, found := c.federatedSeedControllerManager.controllers[newSeedObj.Name]
+	if !found && newSeedBootstrappedCondition != nil && newSeedBootstrappedCondition.Status == gardencorev1beta1.ConditionTrue {
```

Comment:
Hm, why didn't you just fix the bug in the old code? I think it makes sense to only enqueue on update if the condition switches to `True`, or?

Also, the check whether the controller is already started is now duplicated here (the reconciler is already checking it).

/cc @swilen-iwanow @plkokanov @vlvasilev

Reply:
I was thinking that what if we miss an update event? The controller would not be started any more. This way we could even miss the one particular update and would still start the controller at a later point in time. I just find it uncomfortable to rely only on one update event instead just checking if the desired state (having a controller) is fulfilled or not.

Yes the check is duplicated, but it is not an expensive one (lookup key in a map) vs [executing this function](https://github.com/gardener/gardener/blob/46a88c15db5397c573eba6f90f7cdba03327f361/pkg/gardenlet/controller/federatedseed/seed_control.go#L183-L182) and then performing the check.

Reply:
OK, agreed. Makes sense, thanks!